File: tests/fixtures/qualifier_collision.java
// Two classes with method 'process'
public class ProcessorA {
    public void process() {
        System.out.println("ProcessorA");
    }
}

class ProcessorB {
    public void process() {
        System.out.println("ProcessorB");
    }
}

// Caller
class Caller {
    public void run() {
        ProcessorA a = new ProcessorA();
        a.process();  // Should resolve to ProcessorA.process
    }
}

File: tests/fixtures/qualifier_collision.ts
// Two imported modules with same function name
export class ProcessorA {
  process() {
    return "ProcessorA";
  }
}

export class ProcessorB {
  process() {
    return "ProcessorB";
  }
}

// Caller that should use qualified object.method
const processorA = new ProcessorA();
const processorB = new ProcessorB();

function caller() {
  const resultA = processorA.process();  // Should be qualified to processorA.process()
  const resultB = processorB.process();  // Should be qualified to processorB.process()
  return [resultA, resultB];
}
